code/go/0chain.net/chaincore/chain/entity.go

```diff
@@ -575,7 +574,8 @@ func (c *Chain) setupInitialState(initStates *state.InitStates, gb *block.Block)
 		logging.Logger.Debug("init state", zap.String("sc ID", v.ID), zap.Any("tokens", v.Tokens))
 	}
 
-	stateCtx := cstate.NewStateContext(gb, pmt, nil, nil, nil, nil, nil, nil, nil)
+	txn := transaction.Transaction{HashIDField: datastore.HashIDField{Hash: encryption.Hash(c.OwnerID())}, ClientID: c.OwnerID()}
```

Comment:
Txn is not used, please leave it as nil, or show me the place where it's referenced.

Reply:
There is usage for it when we emit events (which otherwise will cause nil pointer error):
https://github.com/0chain/0chain/blob/763a0aa35b0ac2ac47572127bb261753560c560f/code/go/0chain.net/chaincore/chain/state/state_context.go#L330
But the main issue remains the same, the events are not processed. Any help with that will be greatly appreciated.

Reply:
I'll do a quick check tomorrow

Reply:
The events are not processed because the genesis block is finalized when it's created, and because the events are all processed when finalizing blocks, therefore, the events in genesis block would never be processed.